### PR TITLE
Adding integration test for match_delegate + composite + callout_filter in both downstrean and upstream

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
This PR adding integration test to verify match_delegate + composite + callout_filter works in both downstream and upstream. 

It addresses this comments: https://github.com/envoyproxy/envoy/pull/33273#pullrequestreview-1998264437

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/filters/http/ext_proc/config.h [hunk 1]
```diff
@@ -24,7 +24,7 @@ class ExternalProcessingFilterConfig
   static constexpr uint64_t DefaultMessageTimeoutMs = 200;
   static constexpr uint64_t DefaultMaxMessageTimeoutMs = 0;
 
-  virtual absl::StatusOr<Http::FilterFactoryCb> createFilterFactoryFromProtoTyped(
+  absl::StatusOr<Http::FilterFactoryCb> createFilterFactoryFromProtoTyped(
```
Comment (yanjunxiang-google) [inline source/extensions/filters/http/ext_proc/config.h:27]: Make this minor change to keep it same as other dual filters.

**Review by yanavlasov — DISMISSED** [review]

**Review by tyxia — COMMENTED** [review]
LGTM modulo one question in test.

Thanks!

File: test/extensions/filters/http/composite/composite_filter_integration_test.cc [hunk 2]
```diff
@@ -623,49 +688,19 @@ INSTANTIATE_TEST_SUITE_P(
...
-  ASSERT_TRUE(response->waitForEndStream());
-  EXPECT_TRUE(response->complete());
-  EXPECT_EQ(response->headers().getStatusValue(), "200");
+TEST_P(CompositeFilterSeverContextIntegrationTest, BasicFlowDownstreamFilterInDownstream) {
+  initializeConfig(/*downstream_filter*/ true, /*downstream*/ true);
+  serverContextBasicFlowTest();
+}
+
+TEST_P(CompositeFilterSeverContextIntegrationTest, BasicFlowDualFilterInDownstream) {
+  initializeConfig(/*downstream_filter*/ false, /*downstream*/ true);
```
[the remark this answers is not in the document]
Comment (tyxia) [reply test/extensions/filters/http/composite/composite_filter_integration_test.cc:697]: Ah, I think it is for dual filter in downstream filter chain.

Do we still need `BasicFlowDownstreamFilterInDownstream`. In other words, Can we just directly switch  `server_factory_context_filter` to dual filter and remove downstream only version?

I don't feel strongly about this and it is test code.

**Pull request merged**